repose-aggregator/commons/commons-utilities/src/main/scala/org/openrepose/commons/utils/servlet/http/HttpServletRequestWrapper.scala

```diff
@@ -145,4 +150,81 @@ class HttpServletRequestWrapper(originalRequest: HttpServletRequest, inputStream
   def getSplittableHeaderScala(headerName: String): List[String] = getHeadersScala(headerName).foldLeft(List.empty[String])((list, s) => list ++ s.split(","))
 
   override def getSplittableHeaders(headerName: String): util.List[String] = getSplittableHeaderScala(headerName).asJava
+
+  override def getRequestURL: StringBuffer = requestUrl
+
+  def setRequestURL(url: StringBuffer): Unit = {
+    if (Option(url).isEmpty) throw new IllegalArgumentException("null is not a legal argument to setRequestURL")
+
+    requestUrl = url
+  }
+
+  override def getRequestURI: String = requestUri
+
+  def setRequestURI(uri: String): Unit = {
+    if (Option(uri).isEmpty) throw new IllegalArgumentException("null is not a legal argument to setRequestURI")
+
+    requestUri = uri
+  }
+
+  /**
+    * @return a string representation of the query parameters for this request
+    */
+  override def getQueryString: String = queryString
+
+  /**
+    * @param string the desired query string for this request
+    */
+  def setQueryString(string: String): Unit = {
+    // todo: update parameter map
+    // parse the current query string and remove from parameter map
+    // parse the new query string and add to parameter map
+    queryString = string
+  }
+
+  /**
+    * @param key a parameter key
+    * @return the first parameter value associated with the provided key for this request, or null if no value exists
+    */
+  override def getParameter(key: String): String =
+    Option(getParameterValues(key)).map(_.head).orNull
+
+  /**
+    * @param key a parameter key
+    * @return all parameter values associated with the provided key for this request
+    */
+  override def getParameterValues(key: String): Array[String] =
+    parameterMap.map(_.get(key).orNull).getOrElse(super.getParameterValues(key))
+
+  /**
+    * @return all parameter names for this request
+    */
+  override def getParameterNames: util.Enumeration[String] =
+    parameterMap.map(_.keysIterator.asJavaEnumeration).getOrElse(super.getParameterNames)
+
+  /**
+    * @return the parameter map for this request
+    */
+  override def getParameterMap: util.Map[String, Array[String]] =
+    parameterMap.map(_.asJava).getOrElse(super.getParameterMap)
```

Comment:
Apparently I missed that getParameterMap(...) returns an unmodifiable map, as I discovered last week.